src/Compilers/CSharp/Test/Emit3/FlowAnalysis/StructTests.cs

```diff
@@ -219,6 +219,196 @@ public void GenericStructWithPropertyUsingStruct()
+                // (4,13): error CS0523: Struct member 'A<T>.x' of type 'A<A<T>>' causes a cycle in the struct layout
+                //     A<A<T>> x;
+                Diagnostic(ErrorCode.ERR_StructLayoutCycle, "x").WithArguments("A<T>.x", "A<A<T>>").WithLocation(4, 13),
+                // (14,10): error CS0523: Struct member 'C<T>.x' of type 'D<T>' causes a cycle in the struct layout
+                //     D<T> x;
+                Diagnostic(ErrorCode.ERR_StructLayoutCycle, "x").WithArguments("C<T>.x", "D<T>").WithLocation(14, 10),
+                // (18,13): error CS0523: Struct member 'D<T>.x' of type 'C<D<T>>' causes a cycle in the struct layout
+                //     C<D<T>> x;
+                Diagnostic(ErrorCode.ERR_StructLayoutCycle, "x").WithArguments("D<T>.x", "C<D<T>>").WithLocation(18, 13)
+                );
+        }
+
+        [Fact]
+        [WorkItem("https://github.com/dotnet/roslyn/issues/66844")]
+        public void InstanceMemberExplosion_02()
+        {
+            string program = @"#pragma warning disable CS0169 // The field is never used
+struct A<T>
+{
+    A<A<T>> x;
+}
+
+struct B<T>
+{
+    A<C<B<T>>> x;
+}
+
+struct C<T>
+{
+}
+";
+            CreateCompilation(program).VerifyDiagnostics(
+                // (4,13): error CS0523: Struct member 'A<T>.x' of type 'A<A<T>>' causes a cycle in the struct layout
+                //     A<A<T>> x;
+                Diagnostic(ErrorCode.ERR_StructLayoutCycle, "x").WithArguments("A<T>.x", "A<A<T>>").WithLocation(4, 13)
+                );
+        }
+
+        [Fact]
+        [WorkItem("https://github.com/dotnet/roslyn/issues/66844")]
+        public void InstanceMemberExplosion_03()
+        {
+            string program = @"#pragma warning disable CS0169 // The field is never used
+struct E
+{}
+
+struct X<T>
+{
+    T _t;
+}
+
+struct Y
+{
+    X<Z> xz;
+}
+
+struct Z
+{
+    X<E> xe;
+    X<Y> xy;
+}
+";
+            CreateCompilation(program).VerifyDiagnostics(
+                // (12,10): error CS0523: Struct member 'Y.xz' of type 'X<Z>' causes a cycle in the struct layout
+                //     X<Z> xz;
+                Diagnostic(ErrorCode.ERR_StructLayoutCycle, "xz").WithArguments("Y.xz", "X<Z>").WithLocation(12, 10),
+                // (18,10): error CS0523: Struct member 'Z.xy' of type 'X<Y>' causes a cycle in the struct layout
+                //     X<Y> xy;
+                Diagnostic(ErrorCode.ERR_StructLayoutCycle, "xy").WithArguments("Z.xy", "X<Y>").WithLocation(18, 10)
+                );
+        }
+
+        [Fact]
+        [WorkItem("https://github.com/dotnet/roslyn/issues/66844")]
+        public void StaticMemberExplosion_01()
+        {
+            string program = @"#pragma warning disable CS0169 // The field is never used
+struct A<T>
+{
+    static A<A<T>> x;
+}
+
+struct B<T>
+{
+    static A<B<T>> x;
+}
+
+struct C<T>
+{
+    static D<T> x;
+}
+struct D<T>
+{
+    static C<D<T>> x;
+}
```

Comment:
Are we testing a mix of static and instance members? For instance:
```csharp
struct C<T>
{
    static D<T> x;
}
struct D<T>
{
    C<D<T>> x;
}
```

Reply:
> Are we testing a mix of static and instance members? For instance:

These are not interesting in context of the problem we are facing because cycles across static and instance fields are detected separately and by different methods.

Reply:
The example above currently compiles, but peverify fails, and the native compiler reports a cycle. Do we need to log an issue?

Reply:
> Do we need to log an issue?

Please do, we can investigate in more details later..

Reply:
https://github.com/dotnet/roslyn/issues/75832